### PR TITLE
Glitchiness with Tab Characters

### PR DESCRIPTION
**Pull request by justinmc**
Closes https://github.com/flutter/flutter/issues/30173

Entering tab characters into a text field is causing some glitchiness.  It may suddenly cause a single line input to wrap, or the text to flash, etc. at random moments.  Here is a gif from the issue:

It seems like this happens because `mTabWidth` is unassigned and never set, so it contains random values from memory, sometimes zero and sometimes in the billions.  It looks to me like the code that is supposed to set this variable is never called.

My proposed solution in this PR is to hardcode the tab width to 2 and delete the code that's never called.  This fixes the problem for me and seems to give some reasonable behavior for tab characters.  Does anyone else know what the intention of the original code was?  Or any opinions on the width of tabs or what should happen when a tab is entered?

For the record, here is the original commit that added this code in 2015: https://github.com/flutter/engine/commit/01f5266144

### CODE REVIEW

**Comment by jason-simmons** [comment]
This library was originally built to support Android text layout APIs.  Android would call `LineBreaker::setTabStops` and pass a default or user-provided tab stop configuration:
    https://android.googlesource.com/platform/frameworks/base/+/android-6.0.1_r25/core/jni/android_text_StaticLayout.cpp#62

I'd recommend having the `Paragraph` constructor call `breaker_.setTabStops(nullptr, 0, defaultTabStop)`.  Not sure what the default tab stop should be, but 2 seems fine.

**Comment by justinmc** [comment]
Thanks for explaining that @jason-simmons!  I've updated the PR to use your approach and it seems to work just as well.  I'm seeing a tangentially related crash when putting the cursor between tabs that I'm going to investigate before I finalize this PR.

**Comment by justinmc** [comment]
Unfortunately I'm having trouble tracking down the tab crash.  I created a separate issue for it because it existed before this PR and isn't strictly related: https://github.com/flutter/flutter/issues/31154

I'm happy to take a deeper look at it though if someone points me in the right direction.  I'm unfamiliar with what could be causing it and unable to get a full stack trace for now.

**Comment by jason-simmons** [comment]
After taking a look at https://github.com/flutter/flutter/issues/31158, I'm now thinking that it might be better to disable the tab handling logic in the Minikin LineBreaker:
    https://github.com/flutter/engine/blob/master/third_party/txt/src/minikin/LineBreaker.cpp#L162

Flutter doesn't provide the Android StaticLayout tab stop behavior that Minikin's logic appears to be implementing.  Removing the special handling of tabs in LineBreaker results in tabs acting like spaces for the purposes of line breaking and layout.

**Comment by justinmc** [comment]
@jason-simmons Ready for re-review.  I removed the tab code altogether and it does seem to work fine.

File: third_party/txt/src/minikin/LineBreaker.cpp [hunk 1]
```diff
@@ -159,22 +157,13 @@ float LineBreaker::addStyleRun(MinikinPaint* paint,
...
-    } else {
-      if (isWordSpace(c))
-        mSpaceCount += 1;
-      mWidth += mCharWidths[i];
-      if (!isLineEndSpace(c)) {
-        postBreak = mWidth;
-        postSpaceCount = mSpaceCount;
-        afterWord = i + 1;
-      }
+    if (isWordSpace(c))
```
Comment (jason-simmons) [inline third_party/txt/src/minikin/LineBreaker.cpp]: Add a "libtxt:" comment here saying that tab handling has been removed.

(This is intended to keep track of how our version of Minikin differs from the original)

Comment (justinmc) [reply third_party/txt/src/minikin/LineBreaker.cpp]: Got it.  Thanks for the review.

**Pull request merged**